lib/markdownlint.js

```diff
@@ -895,6 +895,39 @@ function parseConfiguration(name, content, parsers) {
   };
 }
 
+/**
+ * Resolve referenced "extends" path ina configuration file
+ * using path.resolve() and require.resolve() as a fallback.
+ *
+ * @param {string} configFile Configuration file name.
+ * @param {*} referenceFile Referenced path to resolve.
+ * @returns {string}
+ */
+function resolveConfigExtends(configFile, referenceFile) {
+  const resolvedExtendsFile = path.resolve(
+    path.dirname(configFile),
+    referenceFile
+  );
+
+  try {
+    if (fs.statSync(resolvedExtendsFile).isFile()) {
```

Comment:
This is used by sync and async callers, right? Goal is to keep sync calls out of the async path.

Reply:
Generally speaking, I agree with the principle. We can duplicate `resolveConfigExtends` so that we have `resolveConfigExtends` and `resolveConfigExtendsSync` like everywhere else, but here are the things to consider:

- `fs.statSync` is super quick, it’s not the same as calling `fs.readFileSync` or `fs.writeFileSync`, which take forever (on CPU scale)
- Node’s `require.resolve` _is_ sync, which in turn depends on some _sync_ fs calls. There can be quite a few of them actually, which involve getting stats different paths, reading `package.json` file and so on.
- Splitting will add more lines to the file, which increases the surface of the bugs
- An async version of the function will make `readConfig` harder to read because we use callbacks and not async / await
- Splitting will require more tests to be added

Happy for you to make a call on this. If a sync nature of `require.resolve` → `fs` is not a sufficient argument for a trade-off I've made, happy to split the function.

Reply:
That's a good point, let's stick with this for now.